cerberus-web/src/main/java/com/nike/cerberus/controller/authentication/AwsIamStsAuthController.java

```diff
@@ -67,24 +68,27 @@ public AuthTokenResponse authenticate(
     String iamPrincipalArn;
     AuthTokenResponse authResponse;
 
-    try {
-      if (headerAuthorization == null || headerXAmzDate == null) {
-        throw new ApiException(DefaultApiError.MISSING_AWS_SIGNATURE_HEADERS);
-      }
-
-      AwsStsHttpHeader header =
-          new AwsStsHttpHeader(headerXAmzDate, headerXAmzSecurityToken, headerAuthorization);
-      GetCallerIdentityResponse getCallerIdentityResponse = awsStsClient.getCallerIdentity(header);
-      iamPrincipalArn = getCallerIdentityResponse.getGetCallerIdentityResult().getArn();
+    for(int count=0;; count++){
+      try {
+        if (headerAuthorization == null || headerXAmzDate == null) {
+          throw new ApiException(DefaultApiError.MISSING_AWS_SIGNATURE_HEADERS);
+        }
+        AwsStsHttpHeader header =
+                new AwsStsHttpHeader(headerXAmzDate, headerXAmzSecurityToken, headerAuthorization);
 
-      authResponse = authenticationService.stsAuthenticate(iamPrincipalArn);
-    } catch (Exception e) {
-      auditLoggingFilterDetails.setAction("Failed to authenticate with AWS IAM STS Auth");
-      throw e;
-    }
+        GetCallerIdentityResponse getCallerIdentityResponse = awsStsClient.getCallerIdentity(header);
+        iamPrincipalArn = getCallerIdentityResponse.getGetCallerIdentityResult().getArn();
 
-    auditLoggingFilterDetails.setAction("Successfully authenticated with AWS IAM STS Auth");
+        authResponse = authenticationService.stsAuthenticate(iamPrincipalArn);
+        auditLoggingFilterDetails.setAction("Successfully authenticated with AWS IAM STS Auth");
 
-    return authResponse;
+        return authResponse;
+      } catch (Exception e) {
+        auditLoggingFilterDetails.setAction("Failed to authenticate with AWS IAM STS Auth");
```

Comment:
should we add the reason the call failed to the audit log?

Reply:
Added